README.md

```diff
@@ -70,106 +86,158 @@ The development version of `libsemigroups` is available on
 [github](https://github.com/libsemigroups/libsemigroups), and some
 related projects are [here](https://github.com/libsemigroups).
 
+## Design principles
+
 The main classes in `libsemigroups` are named after the algorithms they
 implement; see, for example, `libsemigroups::FroidurePin`,
-`libsemigroups::Konieczny`, `libsemigroups::congruence::ToddCoxeter`,
-`libsemigroups::fpsemigroup::Kambites`,
-`libsemigroups::fpsemigroup::KnuthBendix`, and
-`libsemigroups::SchreierSims`, `libsemigroups::Sims1`, or
-`libsemigroups::Stephen`.
-
-The implementations in `libsemigroups::FroidurePin`,
-`libsemigroups::Konieczny`, and `libsemigroups::SchreierSims` are
-generic and easily adapted to user-defined types.
-
-`libsemigroups` uses: [HPCombi](https://github.com/hivert/HPCombi) which
-uses the SSE and AVX instruction sets for very fast manipulation of
-transformations, partial permutations, permutations, and boolean
-matrices of small size; [catch](https://github.com/catchorg/Catch2) for
-tests; [fmt](https://github.com/fmtlib/fmt) for reporting; and
-[eigen](http://eigen.tuxfamily.org/) for some linear algebra
-computations.
+`libsemigroups::Konieczny`, [`libsemigroups::ToddCoxeter`][],
+[`libsemigroups::Kambites`][], [`libsemigroups::KnuthBendix`][],
+`libsemigroups::SchreierSims`, `libsemigroups::Sims1`, `libsemigroups::Sims2`,
+or `libsemigroups::Stephen`.
```

Comment:
Should all of the `libsemigroups::X` occurrences be hyperlinked? Or none of them? Currently looks a bit weird with only some being linked.

Reply:
Actually all of them are linked when this is included in the doc, but none of the links work in the README.md on github.